model_compression_toolkit/core/common/mixed_precision/sensitivity_evaluation.py

```diff
@@ -82,22 +82,40 @@ def __init__(self,
                              f" an HessianInfoService object must be provided but is {hessian_info_service}")
             self.hessian_info_service = hessian_info_service
 
-        # Get interest points for distance measurement and a list of sorted configurable nodes names
         self.sorted_configurable_nodes_names = graph.get_configurable_sorted_nodes_names()
+
+        # Get interest points and output points set for distance measurement and set other helper datasets
+        # We define a separate set of output nodes of the model for the purpose of sensitivity computation.
         self.interest_points = get_mp_interest_points(graph,
                                                       fw_impl.count_node_for_mixed_precision_interest_points,
                                                       quant_config.num_interest_points_factor)
 
-        self.outputs_replacement_nodes = None
-        self.output_nodes_indices = None
-        if self.quant_config.use_grad_based_weights is True:
-            # Getting output replacement (if needed) - if a model's output layer is not compatible for the task of
-            # gradients computation then we find a predecessor layer which is compatible,
-            # add it to the set of interest points and use it for the gradients' computation.
-            # Note that we need to modify the set of interest points before building the models,
-            # therefore, it is separated from the part where we compute the actual gradient weights.
-            self.outputs_replacement_nodes = get_output_replacement_nodes(graph, fw_impl)
-            self.output_nodes_indices = self._update_ips_with_outputs_replacements()
+        self.ips_distance_fns = []
+        self.ips_batch_axis = []
+        for n in self.interest_points:
+            self.ips_distance_fns.append(self.fw_impl.get_node_distance_fn(layer_class=n.layer_class,
+                                                                           framework_attrs=n.framework_attr,
+                                                                           compute_distance_fn=self.quant_config.compute_distance_fn))
+            self.ips_batch_axis.append(n.framework_attr.get(AXIS) if not isinstance(n, FunctionalNode)
+                                       else n.op_call_kwargs.get(AXIS))
+
+        self.output_points = get_output_nodes_for_metric(graph)
+        self.out_ps_distance_fns = []
+        self.out_ps_batch_axis = []
+        for n in self.output_points:
```

Comment:
I think the naming is a bit confusing. Sometimes it's 'ips', sometimes it's 'ps'.
Why don't we just use 'ipt' as an abbreviation in all cases?